File: config-services/src/main/java/com/microservices/configservices/service/InformationService.java
package com.microservices.configservices.service;

import java.util.List;
import java.util.UUID;

import com.microservices.configservices.payload.request.InformationRequest;

import com.microservices.configservices.payload.response.InfomationsResponse;


public interface InformationService {

    InfomationsResponse create(InformationRequest InformationRequest);

    void setImage(UUID id, String image);

    void switchStatus(UUID id);

    void trash(UUID id);

    InfomationsResponse getById(UUID id);

    List<InfomationsResponse> getAll();

    InfomationsResponse update(UUID id, InformationRequest InformationRequest);

    InfomationsResponse delete(UUID id);

    void isDisplay(UUID id);
}

File: reactjs-app/src/services/InformationService.js
import httpAxios from '../httpAxios';

const InformationService = {

    create: (brandData) => {
        return httpAxios.post(`config-services/api/informations/create`, brandData);
    },
    setImage: (data) => {
        return httpAxios.put(`config-services/api/informations/set-image`, data);
    },
    getById: (id) => {
        return httpAxios.get(`config-services/api/informations/get-by-id/${id}`);
    },
    getAll: () => {
        return httpAxios.get(`config-services/api/informations/get-all`);
    },
    update: (id, brand) => { 
        return httpAxios.put(`config-services/api/informations/update/${id}`, brand);
    },
    sitchStatus: (id) => {
        return httpAxios.put(`config-services/api/informations/switch-status/${id}`);
    },
    trash: (id) => {
        return httpAxios.put(`config-services/api/informations/trash/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`config-services/api/informations/delete/${id}`);
    },
    display: (id) => {
        return httpAxios.put(`config-services/api/informations/display/${id}`);
    },
};

export default InformationService;
